packages/speech/src/index.js

```diff
@@ -0,0 +1,816 @@
+ * //     alternatives: [
+ * //       {
+ * //         transcript: 'Result 2b',
+ * //         confidence: 80
+ * //       },
+ * //       ...
+ * //     ]
+ * //   }
+ * // ]
+ */
+Speech.formatResults_ = function(resultSets, verboseMode) {
+  function multiplyScores(result) {
+    if (is.defined(result.confidence)) {
+      result.confidence *= 100;
+    }
+
+    if (is.defined(result.stability)) {
+      result.stability *= 100;
+    }
+
+    return result;
+  }
+
+  var verboseResultSets = resultSets
+    .map(function(resultSet) {
+      resultSet = extend(true, {}, resultSet);
+
+      var mostProbableResult = multiplyScores(resultSet.alternatives.shift());
+
+      resultSet.transcript = mostProbableResult.transcript;
+
+      if (is.defined(mostProbableResult.confidence)) {
+        resultSet.confidence = mostProbableResult.confidence;
+      }
+
+      if (is.defined(mostProbableResult.stability)) {
+        resultSet.stability = mostProbableResult.stability;
+      }
+
+      resultSet.alternatives = resultSet.alternatives.map(multiplyScores);
+
+      return resultSet;
+    });
+
+  if (!verboseMode) {
+    return verboseResultSets.map(prop('transcript')).join(' ');
+  }
+
+  return verboseResultSets;
+};
+
+/**
+ * Perform bidirectional streaming speech-recognition: receive results while
+ * sending audio.
+ *
+ * @resource [`StreamingRecognize` API Reference]{@link https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.Speech.StreamingRecognize}
+ * @resource [`StreamingRecognizeRequest` API Reference]{@link https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognizeRequest}
+ *
+ * @param {object} config - A `StreamingRecognitionConfig` object. See
+ *     [`StreamingRecognitionConfig`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig).
+ * @param {boolean=} config.verbose - Enable verbose mode for a more detailed
+ *     response. See the examples below. Default: `false`.
+ *
+ * @example
+ * var fs = require('fs');
+ *
+ * //-
+ * // See <a href="https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig">
+ * // `RecognizeRequest`</a>.
+ * //-
+ * var request = {
+ *   config: {
+ *     encoding: 'LINEAR16',
+ *     sampleRate: 16000
+ *   },
+ *   singleUtterance: false,
+ *   interimResults: false
+ * };
+ *
+ * fs.createReadStream('./bridge.raw')
+ *   .on('error', console.error)
+ *   .pipe(speech.createRecognizeStream(request))
+ *   .on('error', console.error)
+ *   .on('data', function(data) {
+ *     // The first "data" event emitted might look like:
+ *     //   data = {
+ *     //     endpointerType: Speech.endpointerTypes.START_OF_SPEECH,
+ *     //     results: "",
+ *     //     ...
+ *     //   }
+ *
+ *     // A later "data" event emitted might look like:
```

Comment:
Can we change the comments on the types of data objects returned? Saying that a particular object *might* look like something makes me think that it also might not. We should be providing users with definitive examples of what they should be getting and when they're going to get it.